letsencrypt/configuration.py

```diff
@@ -37,6 +37,11 @@ class NamespaceConfig(object):
 
     def __init__(self, namespace):
         self.namespace = namespace
+
+        self.namespace.config_dir = os.path.abspath(self.namespace.config_dir)
```

Comment:
Really? Why's that?

Sure, the previous solution simplified some of the code in `cli.py`, but that was outside the scope of #836. I think the changes to `configuration.py` are a strict improvement on the old PR.

Previously, I added properties for `cert_path`, `key_path`, `chain_path`, and `fullchain_path`. These should not be properties of the configuration and they are correctly never used that way in the code.

With those changes removed, the only paths that need to be made absolute in `configuration.py` are `config_dir`, `work_dir`, and `logs_dir`. These values do not change during runtime so converting them to an absolute path on every access is unnecessary. Instead, just convert them to an absolute path upfront and all paths based on them will work without further changes.